src/components/application_manager/src/state_controller_impl.cc

```diff
@@ -949,6 +951,15 @@ void StateControllerImpl::OnStateChanged(ApplicationSharedPtr app,
 
   if (new_state->hmi_level() == mobile_apis::HMILevel::HMI_NONE) {
     app->ResetDataInNone();
+    if (mobile_apis::HMILevel::INVALID_ENUM == old_state->hmi_level() &&
+        postponed_activation_controller_.IsAppWaitingForActivation(
+            app->app_id())) {
+      const uint32_t app_id = app->app_id();
+      const uint32_t corr_id =
+          postponed_activation_controller_.GetPendingActivationCorrId(app_id);
+      app_mngr_.ActivatePostponedApp(app_id, corr_id);
+      postponed_activation_controller_.RemoveAppToActivate(app_id);
+    }
```

Comment:
@LitvinenkoIra I believe function `IsAppWaitingForActivation` is redundant and can be removed. Just use corner value of `GetPendingActivationCorrId`:
```suggestion
    if (mobile_apis::HMILevel::INVALID_ENUM == old_state->hmi_level()) {
      const uint32_t app_id = app->app_id();      
      const uint32_t corr_id =
          postponed_activation_controller_.GetPendingActivationCorrId(app_id);
      if (corr_id > 0) {
         app_mngr_.ActivatePostponedApp(app_id, corr_id);
         postponed_activation_controller_.RemoveAppToActivate(app_id);      
      }          
    }
```

Reply:
@AKalinich-Luxoft Done in a2c36cca2cc141e283b4c2fe3444cf724eb4f637